Added description and test plan section in pull request template. (#798)
climateconnect/climateconnect: .github/PULL_REQUEST_TEMPLATE.md

## Description
<!-- Aim to reference the GitHub issue number here (e.g. "#XXX") improve discoverability. -->

> Include summary of the change and list any required dependencies or downstream impacts

Reviewer: ```suggestion
_Include summary of the change and list any required dependencies or downstream impact._
```

> ## Test Plan
> Please include a test plan of new feature, how you verify changes and include relevant test configuration details for the reviewer(s).

Reviewer: ```suggestion
## Test plan
_Include relevant test configuration details for the reviewer(s), and steps to test and verify new feature._
```
